nncf/quantization/algorithms/weight_compression/awq.py

```diff
@@ -0,0 +1,250 @@
+
+from nncf import Dataset
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.graph_matching import find_subgraphs_matching_pattern
+from nncf.common.logging.track_progress import track
+from nncf.common.tensor_statistics.statistic_point import StatisticPointsContainer
+from nncf.experimental.tensor import functions as fns
+from nncf.quantization.algorithms.weight_compression.backend import AWQAlgoBackend
+from nncf.quantization.algorithms.weight_compression.config import WeightCompressionParameters
+from nncf.quantization.algorithms.weight_compression.weight_lowering import do_dequantization
+from nncf.quantization.algorithms.weight_compression.weight_lowering import do_integer_quantization
+from nncf.quantization.passes import transform_to_inference_graph
+
+TModel = TypeVar("TModel")
+TTensor = TypeVar("TTensor")
+TWeightType = TypeVar("TWeightType")
+
+
+class AWQ(AWQAlgoBackend):
+    """
+    Modified AWQ algorithm implementation.
+    """
+
+    def __init__(
+        self,
+        model: TModel,
+        name_to_node_mapping: Dict[str, Any],
+        all_weight_params: List[WeightCompressionParameters],
+        nodes_to_compress: List[NNCFNode],
+        activations: Optional[Dict[str, TTensor]] = None,
+        subset_size: int = 32,
+        percent_to_apply=0.002,
+        alpha_min=0.01,
+        alpha_max=1.0,
+        steps=100,
+    ):
+        """ """
+        super().__init__(model)
+        self.name_to_node_mapping = name_to_node_mapping
+        self._all_weight_params = all_weight_params
+        self._nodes_to_compress = nodes_to_compress
+        self._activations = activations
+        self._subset_size = subset_size
+        self._percent_to_apply = percent_to_apply
+        self._alpha_min = alpha_min
+        self._alpha_max = alpha_max
+        self._steps = steps
+        self._patterns = self.get_awq_patterns()
+
+    def apply(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        statistic_points: Optional[StatisticPointsContainer] = None,
+        dataset: Optional[Dataset] = None,
+    ) -> TModel:
+        """
+        Applies the algorithm to the model.
+
+        :param model: Model for applying algorithm.
+        :param graph: Model graph.
+        :param statistic_points: Statistic points with collected statistics values.
+        :param dataset: A representative dataset for the calibration process.
+        :return: A resulting model.
+        """
+        matches = []
+
+        inference_nncf_graph = transform_to_inference_graph(deepcopy(graph), [], [], [])
+        nx_graph = inference_nncf_graph.get_nx_graph_copy()
+        for _, pattern_graph in self._patterns.items():
+            matches.extend(find_subgraphs_matching_pattern(nx_graph, pattern_graph(), strict=False))
+
+        if len(matches) == 0:
+            return model
+
+        @dataclass
+        class AWQTriplet:
```

Comment:
Adding or removing params would require changing the name.
What about AWQParams? or AWQCompressionInfo?

Reply:
done